src/Hazelcast.Net.BuildAnalyzers/PublicApiNullAttributeAnalyzer.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) 2008-2025, Hazelcast, Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+// http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Hazelcast.Net.BuildAnalyzers
+{
+    /// <summary>
+    /// Checks that all public APIs have either NotNull or MaybeNull parameter. 
+    /// </summary>
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class PublicApiNullAttributeAnalyzer : DiagnosticAnalyzer
```

Comment:
Does Microsoft release any builtin analyzers in .NET framework which we can re-use rather than doing a custom one?

Reply:
I couldn't see such rule unfortunately.